toolchain/check/import_cpp.cpp

```diff
@@ -446,18 +464,33 @@ static auto ImportFunctionDecl(Context& context, SemIR::LocId loc_id,
     context.TODO(loc_id, "Unsupported: Template function");
     return SemIR::ErrorInst::InstId;
   }
-  auto param_patterns_id =
-      MakeParamPatternsBlockId(context, loc_id, *clang_decl);
-  if (!param_patterns_id.has_value()) {
-    return SemIR::ErrorInst::InstId;
-  }
-  auto return_slot_pattern_id = GetReturnType(context, loc_id, clang_decl);
-  if (SemIR::ErrorInst::InstId == return_slot_pattern_id) {
-    return SemIR::ErrorInst::InstId;
+  context.inst_block_stack().Push();
+  context.pattern_block_stack().Push();
+
+  auto decl_block_id = SemIR::InstBlockId::Empty;
+  auto pattern_block_id = SemIR::InstBlockId::Empty;
+  auto param_patterns_id = SemIR::InstBlockId::Empty;
+  auto return_slot_pattern_id = SemIR::InstId::None;
+  auto call_params_id = SemIR::InstBlockId::Empty;
+  {
+    auto scope_exit = llvm::make_scope_exit([&] {
+      decl_block_id = context.inst_block_stack().Pop();
+      pattern_block_id = context.pattern_block_stack().Pop();
+    });
+    param_patterns_id = MakeParamPatternsBlockId(context, loc_id, *clang_decl);
+    if (!param_patterns_id.has_value()) {
+      return SemIR::ErrorInst::InstId;
+    }
+    return_slot_pattern_id = GetReturnType(context, loc_id, clang_decl);
```

Comment:
Do we need to do this after `MakeParamPatternsBlockId()` is called?
Do we need to do this before `CalleePatternMatch()` is called?
Do we need to do this only if  `MakeParamPatternsBlockId()` succeeded?

If not, can we merge this with line 473?
See the tradeoff above.